File: 0001-two-sum/0001-two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}

def two_sum(nums, target)
    num_idx = {}
    nums.each_with_index do |num, i|
        diff = target - num
        if num_idx.key?(diff)
            return [num_idx[diff], i]
        end
        num_idx[num] = i
    end
end

# Basic Solution

def two_sum(nums, target)

    (0...nums.length).each do |i|
        ((i+1)...(nums.length)).each do |j|
            if nums[i] + nums[j] == target
                return [i,j]
            end
        end
    end
end



    

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numIdx = {};
    for (let i = 0; i < nums.length; i++) {
        const diff = target - nums[i];
        if (diff in numIdx) return [numIdx[diff], i];
        numIdx[nums[i]] = i;
    }
};

// Basic Solution

function twoSum(nums, target) {
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[i] + nums[j] === target) {
                return [i,j];
            }
        }
    }
};
